Source/WebCore/rendering/style/StyleCachedImage.cpp

```diff
@@ -83,6 +88,55 @@ URL StyleCachedImage::reresolvedURL(const Document& document) const
     return m_cssValue->reresolvedURL(document);
 }
 
+LegacyRenderSVGResourceContainer* StyleCachedImage::uncheckedRenderSVGResource(TreeScope& treeScope, const AtomString& fragment) const
+{
+    auto renderSVGResource = ReferencedSVGResources::referencedRenderResource(treeScope, fragment);
+    m_isRenderSVGResource = renderSVGResource != nullptr;
+    return renderSVGResource;
+}
+
+LegacyRenderSVGResourceContainer* StyleCachedImage::uncheckedRenderSVGResource(const RenderElement* renderer) const
+{
+    if (!renderer)
+        return nullptr;
+
+    if (imageURL().string().find('#') == notFound) {
```

Comment:
It would be better to use a function on `URL`. Ask Alex which one to use.

Reply:
`URL::hasFragmentIdentifier()` returns false if the url is an inline resource for example `mask-image: url(#starMask);`